src/compiler/types.d.ts

```diff
@@ -8,33 +10,25 @@
  *
  * The return value is set to *, so that we may optionally return any DOM nodes
  * created during a client-side render. These nodes are often needed for ivars.
- *
- * @typedef {function(!Element):*}
  */
-export let BuildDomDef;
+export type BuildDom = (Element) => void;
 
 /**
- * See @ampproject/bento-compiler for the definition.
- * @typedef {*}
+ * Contains component versioning data via a map from tagName --> version.
  */
-export let TreeProtoDef;
+export type Versions = {[tagName: string]: string};
 
 /**
- * Contains component versioning data via a map from tagName --> version.
- *
- * @typedef {Object<string, string>}}
+ * Compiler request
```

Comment:
nit: compiler request/response comments are redundant, can be removed

Reply:
Do you mean that you'd prefer their types inlined in the one location that uses it?

Reply:
Ah, got it!